### PR TITLE
Make `sz test` more performant.

### PR DESCRIPTION
**Pull request by Jonas-Sander**
Tests are now run concurrently for the different packages, instead of only one at a time.  

Time to run tests in GitHub CI:
Old time: 18min 36s
New time: 8-11 min

### CODE REVIEW

**Comment by Jonas-Sander** [comment]
First test run (only `sz test` time, not whole action):

- Concurrency 1: (Still running...)
- Concurrency 3: 7m 13s
- Concurrency 5: 6m 43s 
- Concurrency 7: [FAILED] 10m 6s (`sharezone` app package took longer than 10min --> timeout failure)

**Comment by Jonas-Sander** [comment]
Second test run (only `sz test` time, not whole action):

- Concurrency 3: 8m 19s
- Concurrency 4: [FAILED] 10m 5s (`sharezone` app package took longer than 10min --> timeout failure)
- Concurrency 5: 9m 34s 
- Concurrency 6: 7m 1s

**Comment by Jonas-Sander** [comment]
Third test run (only `sz test` time, not whole action):
The `sharezone` package was now run at the end instead of the beginning.

- Concurrency 3: 8m 19s
- Concurrency 4: [FAILED] 10m 5s (`sharezone` app package took longer than 10min --> timeout failure)
- Concurrency 5: 9m 34s 
- Concurrency 6: 7m 1s

**Comment by Jonas-Sander** [comment]
Third test run (only `sz test` time, not whole action):
The `sharezone` package was now tested at the end instead of the beginning.

- Concurrency 3: 9m 5s
- Concurrency 4: 7m 56s
- Concurrency 5: 8m 1s 
- Concurrency 6: 11m 58s

File: tools/sz_repo_cli/lib/src/commands/src/test_command.dart [hunk 1]
```diff
@@ -7,14 +7,21 @@
...
 class TestCommand extends Command {
-  TestCommand(this._repo) {
+  TestCommand(this.repo) {
     argParser
+      ..addOption(
+        maxConcurrentPackagesOptionName,
+        abbr: 'c',
+        defaultsTo: '5',
+        help:
+            'How many packages at most should be processed at once. Helpful for a CI or not as powerful PCs to not have so much processing at once.',
```
Comment (nilsreichardt) [inline tools/sz_repo_cli/lib/src/commands/src/test_command.dart:23]: Should we also mention that 0 or negative would result into processing all at once?

File: tools/sz_repo_cli/lib/src/commands/src/test_command.dart [hunk 2]
```diff
@@ -25,70 +32,55 @@ class TestCommand extends Command {
...
+      await presenter.printFailedTasksSummary(failures);
+      exit(1);
+    } else {
+      print('All packages tested successfully!');
+      exit(0);
     }
-    print('All Dart package tests passed!');
   }
-}
+}
```
Comment (nilsreichardt) [inline tools/sz_repo_cli/lib/src/commands/src/test_command.dart:86]:
```suggestion
}

```

Nit: Add newline at end of file.

Isn't VS Code adding a newline at the of file when you are pressing auto-format?

**Review by nilsreichardt — APPROVED** [review]
LGTM

File: tools/sz_repo_cli/lib/src/commands/src/test_command.dart (hunk 2 above)
Comment (Jonas-Sander) [reply tools/sz_repo_cli/lib/src/commands/src/test_command.dart]: No locally it adds a new line at the end when I save and the file is auto-formatted.

Comment (Jonas-Sander) [reply tools/sz_repo_cli/lib/src/commands/src/test_command.dart]: For me locally there is a new line at the end - maybe it has something to do with git?

File: tools/sz_repo_cli/lib/src/commands/src/test_command.dart [hunk 3]
```diff
@@ -22,73 +22,59 @@ class TestCommand extends Command {
...
+      print('There were failures. See above for more information.');
+      await presenter.printFailedTasksSummary(failures);
+      exit(1);
+    } else {
+      print('All packages tested successfully!');
+      exit(0);
     }
-    print('All Dart package tests passed!');
   }
 }
```
Comment (Jonas-Sander) [inline tools/sz_repo_cli/lib/src/commands/src/test_command.dart:80]:
```suggestion
}

```

File: tools/sz_repo_cli/lib/src/commands/src/test_command.dart (hunk 2 above)
Comment (Jonas-Sander) [reply tools/sz_repo_cli/lib/src/commands/src/test_command.dart]: I added a new line via GitHub but now there are two for me locally 🤔

**Pull request merged**